server/jetstream_cluster.go

```diff
@@ -2915,25 +2915,30 @@ func (js *jetStream) processClusterCreateConsumer(ca *consumerAssignment, state
 			}
 		}
 
+		update := false
 		// Start our monitoring routine.
 		if rg.node == nil {
 			// Single replica consumer, process manually here.
 			js.processConsumerLeaderChange(o, true)
+			update = true
```

Comment:
But processConsumerLeaderChange() will send something out (see bottom of that function). So it feels like you are going to send 2 responses in some cases?

Reply:
Good catch.. hmm..

Reply:
Need to clear ca.responded but that has to be under js lock.

https://github.com/nats-io/nats-server/blob/c36405ffbc05680d44e8bd75f3a480f1fe2a25ef/server/jetstream_cluster.go#L3345

Reply:
Meaning clear it here before calling into processConsumerLeaderChange..

Reply:
basically ca.responded is never set to false, which is why it is not sent

Reply:
what's the use of this value even? shouldn't that be part of the stack, calling context?

Reply:
Implemented it as Derek suggested. but really, not sure why this variable needs to exist in the first place. 
Shouldn't we always respond when called via `applyMetaEntries` and never when not?

Reply:
Its purpose is to not produce responses after we have already responded.